src/configuration.rs

```diff
@@ -307,12 +308,25 @@ extern "C" fn string_configuration_get<T: ConfigurationValue<ValkeyString> + 'st
     _name: *const c_char,
     privdata: *mut c_void,
 ) -> *mut raw::RedisModuleString {
-    let private_data = unsafe { &*(privdata as *const ConfigrationPrivateData<ValkeyString, T>) };
-    // we know the GIL is held so it is safe to use Context::dummy().
-    private_data
-        .variable
-        .get(&ConfigurationContext::new())
-        .take()
+    match TypeId::of::<T>() {
+        // For ValkeyString type we want to use the method 'safe_clone' in order to not cause a memory leak. Due to this we will do the flow as before explicitly in this match
+        // but instead of doing a clone at the end we do a 'safe_clone'
+        valkeystringtype if valkeystringtype == TypeId::of::<ValkeyGILGuard<ValkeyString>>() => {
+            let private_data = unsafe { &*(privdata as *const ConfigrationPrivateData<ValkeyString, ValkeyGILGuard<ValkeyString>>) };
+            let valkey_gilguard =private_data.variable;
+            let ctx = &ConfigurationContext::new();
+            let value = valkey_gilguard.lock(ctx);
+            value.safer_clone(&Context::dummy()).take()
```

Comment:
This is an alternative to adding a new method of cloning. We can return the same reference for the config get request

```suggestion
            value.safe_clone(&Context::dummy()).inner
```

Would be good to understand the incr / decr flow for config sets and config gets before finalizing on the fix here